### PR TITLE
Housekeeping | Fix note tests

### PR DESCRIPTION
**Pull request by gzurowski**
- None of the note tests were running due to the tests package being non-discoverable.
- Fix existing tests and update assertions to align with the current implementation.

### CODE REVIEW

**Review by yattias — APPROVED** [review]
LGTM

File: src/note/tests/test_note_api.py [hunk 1]
```diff
@@ -86,7 +86,11 @@ def test_cannot_create_shared_note_manually(self):
         # NOTE: Should only be able to created SHARED note by inviting useres
         self.assertNotEqual(created_note['access'], 'SHARED')
 
-    def test_note_editor_cannot_invite_others(self):
+    def test_note_editor_can_invite_others(self):
```
Comment (koutst) [inline src/note/tests/test_note_api.py:89]: @yattias is a note editor supposed to be able to invite others? The old test name suggests they shouldn't be able to.

**Review by koutst — APPROVED** [review]
LGTM

File: src/note/tests/test_note_api.py (hunk 1 above)
Comment (gzurowski) [reply src/note/tests/test_note_api.py:89]: I am not familiar with the business rules around this. I renamed the test to reflect the behavior of the current implementation. It allows for this via the `IsOrganizationUser` permission class on the `invite_user` function.

Comment (koutst) [reply src/note/tests/test_note_api.py:89]: Sorry, this was out of scope of this PR. I was just wondering if we should file a bug to fix later.

**Comment by gzurowski** [comment]
@yattias I am merging this, but it would be good to get your input on https://github.com/ResearchHub/researchhub-backend/pull/1504#discussion_r1541007817.

**Pull request merged**